oak_functions/lookup/src/lib.rs

```diff
@@ -217,6 +217,13 @@ where
         }
     }
 
+    pub fn update_abort(&self) -> UpdateStatus {
+        let mut data_builder = self.data_builder.lock();
+        // Clear the builder throwing away the intermediate result.
+        let _ = data_builder.build();
```

Comment:
I find this a bit counterintuitive (I don't expect a builder to reset itself, and therefore have to call build just to reset). If this is the intention, perhaps build should take ownership of self instead, so that a new builder needs to be explicitly created, and there is no ambiguity about the old one (since it will be gone). Especially since we intentionally need to discard any in progress state (which is the job of the Drop trait).

Reply:
Let's take this discussion and the prioritisation offline. 

Anyway, this is pre-existing and should probably be handled in a different PR, so I'll merge for now and revisit later.